### ISSUE TITLE
[rewrite] Infinite loop in `m.prop.combine` with invalid parameters

### ISSUE THREAD

**Issue by sdougbrown** [lhorie/mithril.js#1307]
### Description:

Passing a non-stream to `m.prop.combine` results in a nasty infinite loop that just keeps throwing: 
```
Uncaught TypeError: Cannot read property 'state' of undefined
```

### Steps to Reproduce:

```
m.prop.combine((prop1, badProp) => {
    ...
}, [m.prop('foo'), <anything else>]);
```

#### Working reproduction

Line 19: http://codepen.io/sdougbrown/pen/rrOWNY

![screen shot 2016-09-09 at 4 10 34 pm](https://cloud.githubusercontent.com/assets/2266707/18401413/5a590928-76a8-11e6-8a11-8590987c06d6.png)

![screen shot 2016-09-09 at 4 11 20 pm](https://cloud.githubusercontent.com/assets/2266707/18401440/73876a48-76a8-11e6-8578-a910d22256c0.png)

### Expected:

`m.prop.combine` should probably throw a more descriptive error and stop further execution.  This can only really happen by mistake, but it can be challenging to figure out what has gone wrong in some cases.

### Actual:

∞

**Comment by isiahmeadows** [comment]
I agree that it should at least check its types to some degree. Silently
swallowing errors is one of the biggest issues people have with promises,
and we really shouldn't reproduce that here.

**Comment by lhorie** [comment]
merged #1309
